TestAnchorExternalHrefIPTimeout is flaky

# Describe the bug
We're trying to rebuild with Go 1.15, but I keep running into issues with `TestAnchorExternalHrefIPTimeout`. 

# To Reproduce
1. `go test github.com/wjdp/htmltest/htmltest`

# Expected behaviour
Tests work.

# Actual behaviour
This randomly fails. Here are [the](https://koji.fedoraproject.org/koji/taskinfo?taskID=47981883) [last](https://koji.fedoraproject.org/koji/taskinfo?taskID=48334441) [four](https://koji.fedoraproject.org/koji/taskinfo?taskID=48413328) [attempts](https://koji.fedoraproject.org/koji/taskinfo?taskID=48413857). As you can see, it fails on different systems every time.

The failure is always the same, though:
```
github.com/wjdp/htmltest/htmltest
<<<<<<<<<<<<<<<<<<<<<<<<
testDocument on ip_timeout.html
fresh --- ip_timeout.html --> http://5.6.7.8
hitting --- ip_timeout.html --> http://5.6.7.8
Get "http://5.6.7.8": dial tcp 5.6.7.8:80: i/o timeout (Client.Timeout exceeded while awaiting headers) --- ip_timeout.html --> http://5.6.7.8
>>>>>>>>>>>>>>>>>>>>>>>>
--- FAIL: TestAnchorExternalHrefIPTimeout (1.00s)
    test_helpers_test.go:23: expected issue request exceeded our ExternalTimeout count 1 != 0
FAIL
```
It seems to be timing out, but raising a slightly different error.

## Versions
 - OS: Fedora Rawhide
 - htmltest: 0.12.1

--- a/htmltest/check-link.go
+++ b/htmltest/check-link.go
@@ -182,18 +182,6 @@ func (hT *HTMLTest) checkExternal(ref *htmldoc.Reference) {
 		<-hT.httpChannel // Bump off http concurrency limiter
 
 		if err != nil {
-			if strings.Contains(err.Error(), "dial tcp") {
-				// Remove long prefix
-				prefix := "Get " + urlStr + ": dial tcp: lookup "
-				cleanedMessage := strings.TrimPrefix(err.Error(), prefix)
-				// Add error
-				hT.issueStore.AddIssue(issues.Issue{
-					Level:     issueLevel,
-					Message:   cleanedMessage,
-					Reference: ref,
-				})
-				return
-			}
 			if strings.Contains(err.Error(), "Client.Timeout") {
 				hT.issueStore.AddIssue(issues.Issue{
 					Level:     issueLevel,
@@ -213,6 +201,20 @@ func (hT *HTMLTest) checkExternal(ref *htmldoc.Reference) {
 					})
 					return
 				}
+			}
+
+			// More generic, should be kept below more specific cases
+			if strings.Contains(err.Error(), "dial tcp") {
+				// Remove long prefix
+				prefix := "Get " + urlStr + ": dial tcp: lookup "
+				cleanedMessage := strings.TrimPrefix(err.Error(), prefix)
+				// Add error
+				hT.issueStore.AddIssue(issues.Issue{
+					Level:     issueLevel,
+					Message:   cleanedMessage,
+					Reference: ref,
+				})
+				return
 			}
 
 			// Unhandled client error, return generic error

--- a/htmltest/test_helpers_test.go
+++ b/htmltest/test_helpers_test.go
@@ -20,7 +20,7 @@ func tExpectIssue(t *testing.T, hT *HTMLTest, message string, expected int) {
 	c := hT.issueStore.MessageMatchCount(message)
 	if c != expected {
 		hT.issueStore.DumpIssues(true)
-		t.Error("expected issue", message, "count", expected, "!=", c)
+		t.Error("expected issue", "'"+message+"'", "count", expected, "!=", c)
 	}
 }
 

